src/internet_identity/src/storage/registration_rates.rs

```diff
@@ -108,20 +106,13 @@ impl<M: Memory> RegistrationRates<M> {
 /// However, because the data is not actually spanning 3 weeks, this underestimates the actual rate.
 /// Taking into account that the data is only spanning 3 days we get the following:
 /// 3 registrations / 259200 seconds = 0.00001157407407 registrations / second
-fn calculate_registration_rate<M: Memory>(
-    now: u64,
-    data: &MinHeap<Timestamp, M>,
-    data_retention_ns: u64,
-) -> f64 {
+fn calculate_registration_rate<M: Memory>(now: u64, data: &MinHeap<Timestamp, M>) -> f64 {
     data
-        // get the oldest expiration timestamp
+        // get the oldest value
         .peek()
-        // calculate the registration timestamp from expiration
-        .map(|ts| ts - data_retention_ns)
         // calculate the time window length with respect to the current time
         .map(|ts| now - ts)
-        // the value _could_ be 0 if the oldest timestamp was added in the same execution round
-        // -> filter to avoid division by 0
+        // There could be negative values if they haven't been pruned yet.
```

Comment:
The values are `u64` so the cannot be negative. But also `ts` is never bigger than `now` so the minimum value is 0, no?

Reply:
Oh, sorry, yes. I did some changes here and I didn't realize that it doesn't make sense this anymore.